inc/views/base/calendar/addevent.twig

```diff
@@ -0,0 +1,404 @@
+            </tr>
+            <tr>
+                <td width="20%" class="trow1"><strong>{{ lang.event_name }}</strong></td>
+                <td class="trow1"><input type="text" class="textbox" size="50" name="name" maxlength="120" value="{{ select.name }}"/></td>
+            </tr>
+            {% if calendarcount > 1 %}
+                <tr>
+                    <td width="20%" class="trow1"><strong>{{ lang.select_calendar }}</strong></td>
+                    <td class="trow1">
+                        <select name="calendar">
+                            {% for calendar in calendar_select %}
+                                <option value="{{ calendar.cid }}"{{ calendar.selected }}>{{ calendar.name }}</option>
+                            {% endfor %}
+                        </select>
+                    </td>
+                </tr>
+            {% else %}
+                <input type="hidden" name="calendar" value="{{ calendar.cid }}" />
+            {% endif %}
+            <tr>
+                <td class="trow2" style="vertical-align: top;"><strong>{{ lang.event_date }}</strong></td>
+                <td class="trow1">
+                    <script type="text/javascript">
+                    <!--
+                        function toggleType()
+                        {
+                            if($("#type_single").prop("checked"))
+                            {
+                                $("#ranged_selects").hide();
+                                $("#single_selects").show();
+                            }
+                            else
+                            {
+                                $("#single_selects").hide();
+                                $("#ranged_selects").show();
+                                toggleRepeats();
+                            }
+                        }
+
+                        function toggleRepeats()
+                        {
+                            var repeats = $("#repeats").get(0).selectedIndex;
+                            $(".repeats_type").each(function()
+                            {
+                                $(this).hide();
+                            });
+                            if($("#repeats_type_"+repeats).length > 0)
+                            {
+                                $("#repeats_type_"+repeats).show();
+                            }
+                        }
+                    // -->
+                    </script>
+                    <dl style="margin-top: 0; margin-bottom: 0; width: 100%;">
+                        <dt><input type="radio" name="type" value="single" {{ select.type_single }} id="type_single" onclick="toggleType();" /> <label for="type_single"><strong>{{ lang.type_single }}</strong></label></dt>
+                        <dd style="margin-top: 4px;" id="single_selects">
+                            <select name="single_day">
```

Comment:
Perhaps a good idea to try and abstract this into a loop and possibly break it into a function or partial view or something?